### DISCUSSION TITLE
How to add document reference to firestore document references (array) using Rowy?

### DISCUSSION THREAD

**Discussion by ortsuev** [AntlerVC/firetable#1094] [Support / Q&A] [answered]
I have a list of references to documents that I store in a corresponding array in firestore. I need to use the rowy extensions to create a new document when I get a webhook, and then put a reference to it in that document array. As we know, rowy does not yet support reference arrays as a type. Can I solve my problem by using functions? I'd be happy to get any help

**Comment by bofeiw** [comment]
As an alternative, you can use JSON to store array data. Does this help your use case?

<img width="465" alt="Screenshot 2023-01-26 at 00 54 51" src="https://user-images.githubusercontent.com/34177142/214643868-56b081c4-fae6-42f1-aba6-693c0c16f718.png">

**Reply by ortsuev** [reply]
Hi @bofeiw ! Thank you for the answer! When I choose a column type like JSON (array) for my firestore's document references array -  it shows me something like this 
<img width="460" alt="image" src="https://user-images.githubusercontent.com/123541068/214822122-21dd44ff-dfb5-49dc-becd-fea4b5d6dcc6.png">

**Reply by ortsuev** [reply]
I dont understand how to achive it shows document references like you showed  (

**Reply by bofeiw** [reply]
Hi, please do not store the Firebase Document object directly. One possible way is to convert the object to path strings.
E.g. `doc.ref.path`
https://stackoverflow.com/a/57764840/12208834

**Comment by harinij** [comment, marked as answer]
Hi @ortsuev you can use the newly introduced Array field for this now.

Documentation: https://docs.rowy.io/field-types/array
Video: https://youtu.be/-_zQjsIwPx8
